lib/python/dataset.cpp

```diff
@@ -279,6 +286,7 @@ Returned by :py:func:`DataArray.masks`)");
   bind_binary<Dataset>(dataset);
   bind_binary<DataArray>(dataset);
   bind_binary<Variable>(dataset);
+  bind_binary_scalars(dataset);
```

Comment:
This showed up because `Dataset.__add__` and `Dataset.__iadd__` were inconsistent. Only the latter supported floats on the RHS.

There are still inconsistencies in `Variable` and `DataArray` because they allow binary ops with `Dataset` but not in-place ops. This makes sense. But I think the clean solution would be, e.g., to not implement `Variable.__add__(Dataset)` but instead `Dataset.__radd__(Variable)`. I can do that but I don't know how much of a refactor that would be. So I didn't.